drivers/pinctrl/pinctrl_stm32.c

```diff
@@ -87,6 +87,47 @@ SYS_INIT(stm32_pinmux_init_remap, PRE_KERNEL_1,
 
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
 
+#if DT_NODE_HAS_PROP(DT_NODELABEL(pinctrl), swj_cfg)
+
+#if (DT_ENUM_IDX(DT_NODELABEL(pinctrl), swj_cfg) == 0)
+#define PINCTRL_SWJ_CFG_NONJTRST 1
+#elif (DT_ENUM_IDX(DT_NODELABEL(pinctrl), swj_cfg) == 1)
+#define PINCTRL_SWJ_CFG_JTAGDISABLE 1
+#elif (DT_ENUM_IDX(DT_NODELABEL(pinctrl), swj_cfg) == 2)
+#define PINCTRL_SWJ_CFG_DISABLE 1
+#else
+#error Invalid swj-cfg
+#endif
```

Comment:
Not blocking: we could use directly `(DT_ENUM_IDX(DT_NODELABEL(pinctrl), swj_cfg) == 0)` in the function below.